technico-v16/app/user/user-repairs/page.tsx

```diff
@@ -0,0 +1,135 @@
+"use client";
+
+import { useAuth } from "@/app/components/AuthContext";
+import RepairCard from "@/app/components/RepairCard";
+import { useEffect, useState } from "react";
+
+interface Repair {
+    repairId: string;
+    scheduledDate: Date;
+    type: string;
+    description: string;
+    address: string;
+    status: string;
+    cost: number;
+    userId: string;
+    propertyId: string;
+}
+
+const UserRepairs = () => {
+    const [repairs, setRepairs] = useState<Repair[]>([]);
+    const [filteredRepairs, setFilteredRepairs] = useState<Repair[]>([]);
+    const [searchQuery, setSearchQuery] = useState<string>('');
+    const [loading, setLoading] = useState(true);
+    const [error, setError] = useState<string | null>(null);
+    const { isAuthenticated } = useAuth();
+
+    useEffect(() => {
+        fetchRepairs();
+    }, [isAuthenticated]);
+
+    useEffect(() => {
```

Comment:
Good use of a computed property to filter repairs dynamically based on searchQuery.